File: week3/hayeounglee/BOJ_덱.py
import sys
from collections import deque

input = sys.stdin.readline
n = int(input())

dq = deque()

for _ in range(n):
    cmd = input().split()

    if cmd[0] == "push_front":
        dq.appendleft(cmd[1])

    elif cmd[0] == "push_back":
        dq.append(cmd[1])

    elif cmd[0] == "pop_front":
        print(dq.popleft() if dq else -1)

    elif cmd[0] == "pop_back":
        print(dq.pop() if dq else -1)

    elif cmd[0] == "size":
        print(len(dq))

    elif cmd[0] == "empty":
        print(0 if dq else 1)

    elif cmd[0] == "front":
        print(dq[0] if dq else -1)

    elif cmd[0] == "back":
        print(dq[-1] if dq else -1)

File: week3/sohee/BOJ_덱.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    int n, i;
    cin >> n;
    string s;
    deque<int> dq;
    while(n--){
        cin >> s;
        if (s == "push_back") {
            cin >> i;
            dq.push_back(i);
        } else if (s == "push_front") {
            cin >> i;
            dq.push_front(i);
        } else if (s == "front") {
            if (dq.size()) cout << dq.front() << "\n";
            else cout << -1 << "\n";
        } else if (s == "back") {
            if (dq.size()) cout << dq.back() << "\n";
            else cout << -1 << "\n";
        } else if (s == "size") {
            cout << dq.size() << "\n";
        } else if (s == "empty") {
            cout << (dq.size() == 0 ? 1 : 0) << "\n";
        } else if (s == "pop_front") {
            if (dq.size()) {
                cout << dq.front() << "\n";
                dq.pop_front();
            } else {
                cout << -1 << "\n";
            }
        } else if (s == "pop_back") {
            if (dq.size()) {
                cout << dq.back() << "\n";
                dq.pop_back();
            } else {
                cout << -1 << "\n";
            }
        } 
    }
}